src/SimCore/ReSimulator.cxx

```diff
@@ -9,23 +9,50 @@ void ReSimulator::configure(framework::config::Parameters& parameters) {
   if (!resimulate_all_events) {
     events_to_resimulate_ =
         parameters.getParameter<std::vector<int>>("events_to_resimulate", {});
+    runs_to_resimulate_ = 
+        parameters.getParameter<std::vector<int>>("runs_to_resimulate", {});
     if (events_to_resimulate_.size() == 0) {
       EXCEPTION_RAISE(
           "ReSimNoEvents",
           "ReSim was configured with resimulate_all_events marked false but "
           "no event numbers were requested.\n\nDid you forget to configure "
           "the events_to_resimulate parameter?\n");
     }
+    if (runs_to_resimulate_.size() > 1 and runs_to_resimulate_.size() != events_to_resimulate_.size()) {
+      EXCEPTION_RAISE(
+          "ReSimBadConf",
+          "ReSim needs the runs list to match the length of the events list if there is more than "
+          "one run that needs to be resimulated.");
+    }
   }
 }
 void ReSimulator::produce(framework::Event& event) {
   /* numEventsBegan_++; */
   auto& eventHeader{event.getEventHeader()};
   const auto eventNumber{eventHeader.getEventNumber()};
   if (!resimulate_all_events) {
-    auto found = std::find(std::begin(events_to_resimulate_),
-                           std::end(events_to_resimulate_), eventNumber);
-    if (found == std::end(events_to_resimulate_)) {
+    auto found_event = std::find(std::begin(events_to_resimulate_),
```

Comment:
I feel like the logic here could be simpler, it is a bit hard to read (at least for me) here, which is why we need so many comments. I think we could create a similar variable to resimulate_all_events that we set in the configuration that tells us what behaviour we are expecting from the run numbers.

Reply:
If there isn't a straight-forward way to simplify the logic here, I would advocate for creating a vector of event/run pairs during configuration, possibly as a separate function call if it is > a few lines, (encapsulating the messy logic there) and simply do a find here on that vector. In case 2, this vector would just have all the event numbers in events_to_resimulate_ paired with the one run number we care about.

Reply:
I like creating the pairs in `configure` to make it really clear to future developers how we are interpreting the configuration. As a sanity check, I double checked that `std::find` can work over `std::pair` and it does: https://godbolt.org/z/v4T7a4fKh